### ISSUE TITLE
Failed: MailboxSelectorSpec

### ISSUE THREAD

**Issue by johanandren** [akka/akka#30363]
(on Scala 3 but I think that is unrelated)

https://app.travis-ci.com/github/akka/akka/jobs/521145928

```
-> [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must set capacity on a bounded mailbox] Start of log messages of test that [Failed(java.lang.AssertionError: Received 1 excess messages on LoggingTestKitImpl(1,Some(INFO),None,None,None,Some(.*was not delivered.*dead letters encountered.*),None,Map(),None).)]
38972021-07-02 10:29:18,702 INFO  akka.event.slf4j.Slf4jLogger  - Slf4jLogger started MDC: {}
38982021-07-02 10:29:18,708 DEBUG MailboxSelectorSpec  - Starting ActorTestKit MDC: {}
38992021-07-02 10:29:18,712 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging started for test [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must default to unbounded] MDC: {}
39002021-07-02 10:29:18,716 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging finished for test [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must default to unbounded] that [Succeeded] MDC: {}
39012021-07-02 10:29:18,717 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging started for test [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must select a bounded mailbox] MDC: {}
39022021-07-02 10:29:18,718 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging finished for test [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must select a bounded mailbox] that [Succeeded] MDC: {}
39032021-07-02 10:29:18,719 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging started for test [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must set capacity on a bounded mailbox] MDC: {}
39042021-07-02 10:29:18,721 INFO  akka.actor.LocalActorRef akkaDeadLetter - Message [java.lang.String] to Actor[akka://MailboxSelectorSpec/user/$c#-1976620140] was not delivered. [1] dead letters encountered. If this is not an expected behavior then Actor[akka://MailboxSelectorSpec/user/$c#-1976620140] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'. MDC: {akkaAddress=akka://MailboxSelectorSpec, sourceThread=MailboxSelectorSpec-akka.actor.default-dispatcher-6, akkaSource=akka://MailboxSelectorSpec/user/$c, sourceActorSystem=MailboxSelectorSpec, akkaMessageClass=java.lang.String, akkaTimestamp=10:29:18.721UTC}
39052021-07-02 10:29:18,722 INFO  akka.actor.LocalActorRef akkaDeadLetter - Message [java.lang.String] to Actor[akka://MailboxSelectorSpec/user/$c#-1976620140] was not delivered. [2] dead letters encountered. If this is not an expected behavior then Actor[akka://MailboxSelectorSpec/user/$c#-1976620140] may have terminated unexpectedly. This logging can be turned off or adjusted with configuration settings 'akka.log-dead-letters' and 'akka.log-dead-letters-during-shutdown'. MDC: {akkaAddress=akka://MailboxSelectorSpec, sourceThread=MailboxSelectorSpec-akka.actor.default-dispatcher-6, akkaSource=akka://MailboxSelectorSpec/user/$c, sourceActorSystem=MailboxSelectorSpec, akkaMessageClass=java.lang.String, akkaTimestamp=10:29:18.721UTC}
39062021-07-02 10:29:18,822 INFO  akka.actor.testkit.typed.scaladsl.LogCapturing  - Logging finished for test [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must set capacity on a bounded mailbox] that [Failed(java.lang.AssertionError: Received 1 excess messages on LoggingTestKitImpl(1,Some(INFO),None,None,None,Some(.*was not delivered.*dead letters encountered.*),None,Map(),None).)] MDC: {}
3907<-- [akka.actor.typed.MailboxSelectorSpec: The Mailbox selectors must set capacity on a bounded mailbox] End of log messages of test that [Failed(java.lang.AssertionError: Received 1 excess messages on LoggingTestKitImpl(1,Some(INFO),None,None,None,Some(.*was not delivered.*dead letters encountered.*),None,Map(),None).)]
3908[info] - must set capacity on a bounded mailbox *** FAILED *** (108 milliseconds)
3909[info]   java.lang.AssertionError: Received 1 excess messages on LoggingTestKitImpl(1,Some(INFO),None,None,None,Some(.*was not delivered.*dead letters encountered.*),None,Map(),None).
3910[info]   at akka.actor.testkit.typed.internal.LoggingTestKitImpl.expect(LoggingTestKitImpl.scala:103)
3911[info]   at akka.actor.typed.MailboxSelectorSpec.f$proxy3$2(MailboxSelectorSpec.scala:82)
3912[info]   at akka.actor.typed.MailboxSelectorSpec.$init$$$anonfun$4$$anonfun$3(MailboxSelectorSpec.scala:65)
3913[info]   at org.scalatest.Transformer.apply$$anonfun$1(Transformer.scala:22)
3914[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
3915[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:31)
3916[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
3917[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
3918[info]   at org.scalatest.Transformer.apply(Transformer.scala:21)
3919[info]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1105)
```

**Comment by raboof** [comment]
Seen again in https://github.com/akka/akka/pull/30415 (again scala3)

**Comment by jlprat** [comment]
Failed here: https://github.com/akka/akka/pull/30433 (also scala3)

**Comment by raboof** [comment]
also https://github.com/akka/akka/actions/runs/1068321962

**Comment by octonato** [comment]
Fix in #30429
